src/status_im/contexts/profile/login/events.cljs

```diff
@@ -142,6 +141,7 @@
            (when (and new-account? (not recovered-account?))
              [:dispatch [:wallet-legacy/set-initial-blocks-range]])
            [:dispatch [:ens/update-usernames ensUsernames]]
+           [:dispatch [:wallet/initialize]]
```

Comment:
Initialize wallet earlier so that we have the wallet cards ready to be in loading state when the user lands on the home screen

Reply:
cc @smohamedjavid

Reply:
@OmarBasem this is what I was referring to in your previous PR when I mentioned the `ready?` flag  or changing the subscription to check for `nil` values and take it as "loading".

Now we are initializing the wallet earlier, IMO doing this just to have a loading state is not great, instead we could modify the subscription.

Reply:
how could the modified subscription look/act @ulisesmac ? 🤔

Reply:
> this is what I was referring to in your previous PR when I mentioned the ready?

@ulisesmac I am not sure what does the `tokens-loading?` has to do with the issue here? The problem here is that token initialize takes a little time and we don't know yet how many cards to display as loading. When we initialize a little earlier we know how many cards to display as loading

Reply:
@OmarBasem  OK!, thanks for the clarification. Yes, the loading state is not related.

> When we initialize a little earlier, we get the data on time, and we know how many cards to display as loading

Doesn't it depends on the network and maybe we still have the same issue?

Reply:
> Doesn't it depends on the network and maybe we still have the same issue?

@ulisesmac If the network was very slow then the cards may be delayed for a split second. We can ask design team for a design for that state like grey cards, but probably not a priority now.

Reply:
> Initialize wallet earlier so that we have the wallet cards ready to be in loading state when the user lands on the home screen

As for the loading state, we already set the `token-loading? true` when app-db is initialized.

https://github.com/status-im/status-mobile/blob/0d6bd9c15df6e985f01903d44478241a1a0cfe7c/src/status_im/db.cljs#L22


> When we initialize a little earlier, we get the data on time, and we know how many cards to display as loading

The `:wallet/get-accounts` inside `:wallet/initialize` will be faster (just a read/fetch from DB) but `:wallet/get-wallet-tokens` will be slower as it includes the updated tokens balances and market values for the user's preferred currency. There is no guarantee that the request will be faster and the user will see loaded tokens upon landing in the wallet tab. If the account(s) has no tokens, it will be faster but if the account(s) has some tokens, it will take some time. It depends on the network as well.

IMO, the loading state for tokens in the wallet apps is normal after login.

Reply:
@OmarBasem 
OK! and do you still think it's worth moving the wallet initialization?

Does moving the initialization has a side effect? like making the login slower or something?

Reply:
do we need to batch the wallet initilization?
i.e split out certain parts? like get account data immediately but get token data slightly later? 🤔

Reply:
I will say the first impression of the wallet seems a lot smoother in this video. Nice one @OmarBasem! :)

Reply:
> As for the loading state, we already set the token-loading? true when app-db is initialized.

@smohamedjavid the problem in this PR is not about the loading state, but not having the account cards to display as loading, so what happens is like the below screenshot

<img width="375" alt="Screenshot 2024-06-06 at 09 18 06" src="https://github.com/status-im/status-mobile/assets/29354102/ee97a9c2-b1bc-4f29-b67f-c6c95a227b6c">

> The :wallet/get-accounts inside :wallet/initialize will be faster (just a read/fetch from DB)

what do you mean by that? I see that `:wallet/get-accounts` is an rpc call.


> @OmarBasem
OK! and do you still think it's worth moving the wallet initialization?

@ulisesmac Yeah I think moving the wallet initialization makes it look smoother as @J-Son89 said

FYI guys, there are still further optimizations that can be done to this initialization regarding loading time as described in the following issue https://github.com/status-im/status-mobile/issues/20275

Reply:
@J-Son89 - That's how it's happening right now. All the fetching tokens, and collectibles and checking recent history are happening upon success call of get-accounts.

> @smohamedjavid the problem in this PR is not about the loading state, but not having the account cards to display as loading, so what happens is like the below screenshot (you can also https://github.com/status-im/status-mobile/pull/20297#issue-2330187160 here)

Thanks for the explanation @OmarBasem  🙏 

I see what you mean. Since we take the user to the shell tab immediately after login (while we wait for the account's data), this issue is most likely to be experienced. There is no loading screen after login like the desktop app which gives a user some UI to look at while we prepare the data for them.

Additionally, I looked at the issue https://github.com/status-im/status-mobile/issues/20275, the accounts data we get from status-go is the bare minimal data. We can look into other areas such as assets for each account to check for performance bottlenecks. 👍